File: src/main/java/core/gui/DebugMode.java
package core.gui;

import core.db.frontend.SQLDialog;
import core.net.Connector;

import javax.swing.*;
import java.awt.event.ItemEvent;

public final class DebugMode {

	private DebugMode() {
		throw new UnsupportedOperationException("This is a utility class and cannot be instantiated");
	}

	public static JMenu getDeveloperMenu() {
		JMenu menu = new JMenu("Debug");
		menu.add(createSqlDialogMenuItem());
		menu.add(createLookAndFeelDialogMenuItem());
		menu.add(createSaveDownloadedXmlMenuItem());
		return menu;
	}

	private static JMenuItem createLookAndFeelDialogMenuItem() {
		JMenuItem newItem = new JMenuItem("Look and Feel");
		newItem.addActionListener(e -> new LookAndFeelDialog().setVisible(true));
		return newItem;
	}

	private static JMenuItem createSqlDialogMenuItem() {
		JMenuItem newItem = new JMenuItem("SQL Editor");
		newItem.addActionListener(e -> new SQLDialog().setVisible(true));
		return newItem;
	}

	private static JMenuItem createSaveDownloadedXmlMenuItem() {
		JMenuItem newItem = new JCheckBoxMenuItem("Save downloaded XML", Connector.isSaveDownloadedXml());
		newItem.addItemListener(e -> Connector.setSaveDownloadedXml(e.getStateChange() == ItemEvent.SELECTED));
		return newItem;
	}
}

File: src/main/java/core/gui/DebugMode.kt
package core.gui

import core.db.frontend.SQLDialog
import core.net.MyConnector
import java.awt.event.ActionEvent
import java.awt.event.ItemEvent
import javax.swing.JCheckBoxMenuItem
import javax.swing.JMenu
import javax.swing.JMenuItem

object DebugMode {
    val developerMenu: JMenu
        get() {
            val menu = JMenu("Debug")
            menu.add(sQLDialogMenuItem)
            menu.add(lookAndFeelDialogMenuItem)
            menu.add(saveXMLMenuItem)
            return menu
        }
    private val lookAndFeelDialogMenuItem: JMenuItem
        get() {
            val newItem = JMenuItem("Look and Feel")
            newItem.addActionListener { e: ActionEvent? -> LookAndFeelDialog().isVisible = true }
            return newItem
        }
    private val sQLDialogMenuItem: JMenuItem
        get() {
            val newItem = JMenuItem("SQL Editor")
            newItem.addActionListener { e: ActionEvent? -> SQLDialog().isVisible = true }
            return newItem
        }
    private val saveXMLMenuItem: JMenuItem
        get() {
            val newItem: JMenuItem = JCheckBoxMenuItem("Save downloaded XML")
            newItem.addItemListener { e: ItemEvent -> MyConnector.setDebugSave(e.stateChange == ItemEvent.SELECTED) }
            return newItem
        }
}